### ISSUE TITLE
[request] suitesparse/5.10.1

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan-center-index#5763]
### Package Details
  * Package Name/Version: **suitesparse/5.10.1**
  * Website: **https://people.engr.tamu.edu/davis/suitesparse.html**
  * Source code: **https://github.com/DrTimothyAldenDavis/SuiteSparse**

### Description Of The Library / Tool

SuiteSparse is a suite of sparse matrix algorithms.

**Comment by ericriff** [comment]
Related: https://github.com/conan-io/conan-center-index/blob/master/recipes/ceres-solver/all/conanfile.py#L103

**Comment by sxsong1207** [comment]
Yeah, that would be helpful. Without SUITESPARSE, the covariance computation of ceres-solver is extremely slow.

https://github.com/ceres-solver/ceres-solver/issues/766#issuecomment-1068653277

**Comment by valgur** [comment]
SuiteSparse is ready for packaging, now that we have a working OpenBLAS with BLAS and LAPACK support and an almost fully-functional OpenMP support (#22353).

As the name suggests, SuiteSparse consists of a suite of mostly independent libraries, where each of them:
* has a distinct versioning scheme,
* varying license terms (e.g. some are GPL, while others are more permissive),
* installs a separate CMake config file (i.e. you would do `find_package(AMD)` to use `suitesparse-amd`),
* can be implemented in pure C, while some use C++ as well,
* possibly has library-specific options, e.g. for CUDA support.

So it makes more sense to package each of the sub-components separately, despite it requiring the creation of a considerable number of new libraries and related PRs. This also allows only specific components to be used as needed with available binaries, without having to build everything (such as suitesparse-graphblas, which can take up to an hour to build with pre-built kernels enabled).

Here's the list of sub-packages I intend to create a PR for shortly:
- suitesparse-config
- suitesparse-amd
- suitesparse-btf
- suitesparse-camd
- suitesparse-ccolamd
- suitesparse-cholmod
- suitesparse-colamd
- suitesparse-cxsparse
- #23320
- suitesparse-klu
- suitesparse-lagraph
- suitesparse-ldl
- suitesparse-mongoose
- suitesparse-paru
- suitesparse-rbio
- suitesparse-spex
- suitesparse-spqr
